Client/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```diff
@@ -62,353 +67,146 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Blip", "Element");
 }
 
-int CLuaBlipDefs::CreateBlip(lua_State* luaVM)
+std::variant<CClientRadarMarker*, bool> CLuaBlipDefs::CreateBlip(lua_State* luaVM, CVector vecPosition, std::optional<std::uint8_t> icon, std::optional<std::uint8_t> size, std::optional<std::uint8_t> r, std::optional<std::uint8_t> g, std::optional<std::uint8_t> b, std::optional<std::uint8_t> a, std::optional<std::int32_t> ordering, std::optional<std::uint32_t> visibleDistance)
 {
-    CVector          vecPosition;
-    unsigned char    ucIcon = 0;
-    int              iSize = 2;
-    SColorRGBA       color(255, 0, 0, 255);
-    int              iOrdering = 0;
-    int              iVisibleDistance = 16383;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadVector3D(vecPosition);
-    argStream.ReadNumber(ucIcon, 0);
-    argStream.ReadNumber(iSize, 2);
-    argStream.ReadNumber(color.R, 255);
-    argStream.ReadNumber(color.G, 0);
-    argStream.ReadNumber(color.B, 0);
-    argStream.ReadNumber(color.A, 255);
-    argStream.ReadNumber(iOrdering, 0);
-    argStream.ReadNumber(iVisibleDistance, 16383);
-
-    if (!CClientRadarMarkerManager::IsValidIcon(ucIcon))
+    if (icon.has_value() && !CClientRadarMarkerManager::IsValidIcon(icon.value()))
+        throw std::invalid_argument("Invalid icon");
+
+    if (size.has_value() && (size.value() < 0 || size.value() > 25))
     {
-        argStream.SetCustomError("Invalid icon");
+        m_pScriptDebugging->LogWarning(luaVM, SString("Blip size beyond 25 is no longer supported (got %i). It will be clamped between 0 and 25.", size.value()));
+        size = Clamp(0ui8, size.value(), 25ui8);
     }
 
-    if (iSize < 0 || iSize > 25)
-        argStream.SetCustomWarning(SString("Blip size beyond 25 is no longer supported (got %i). It will be clamped between 0 and 25.", iSize));
+    if (ordering.has_value())
+        ordering = Clamp(std::numeric_limits<std::int16_t>().min(), (std::int16_t)ordering.value(), std::numeric_limits<std::int16_t>().max());
```

Comment:
Could be also represented as `Clamp<std::int16_t>(std::numeric_limits<std::int16_t>().min(), ordering.value(), std::numeric_limits<std::int16_t>().max());`. Or even maybe `std::clamp(...)`. No pressure, just speaking about some variants.